src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -795,14 +835,402 @@ def __try_add_error(error_list, detail):
+                                    __create_installation_tombstone
+                                    __recreate_installation_summary_json()
+                                    
+                            current_package_list_size -= diff
+                            if current_package_list_size <= max_package_list_capacity:
+                                break
+                            __get_new_size_in_bytes_after_truncation() 
+                    
+                write_with_retry_using_temp_file()  
+        """
+        status_file_size_in_bytes = len(complete_status_file_payload.encode("utf-8"))   # calc complete_status_file_payload byte size on disk
+        truncated_status_file = json.loads(complete_status_file_payload)    # reload payload into python object
+
+        if status_file_size_in_bytes > self.__internal_file_capacity:   # 126kb
+            self.composite_logger.log_debug("Begin package list truncation")
+            low_pri_index = None
+            packages_capacity_diff = 0
+            assessment_substatus_index = self.__get_substatus_index(Constants.PATCH_ASSESSMENT_SUMMARY, truncated_status_file['status']['substatus'])
+            installation_substatus_index = self.__get_substatus_index(Constants.PATCH_INSTALLATION_SUMMARY, truncated_status_file['status']['substatus'])
+
+            if assessment_substatus_index is not None:        # if assessment data exists
+                self.__assessment_substatus_msg_copy = self.__get_substatus_msg_at_index(truncated_status_file, assessment_substatus_index)
+                self.__assessment_packages_copy = self.__assessment_substatus_msg_copy['patches']
+
+            if installation_substatus_index is not None:      # if installation data exists
+                low_pri_index = self.__get_installation_low_pri_index(self.__installation_packages_copy, low_pri_index)
+                self.__installation_substatus_msg_copy = self.__get_substatus_msg_at_index(truncated_status_file, installation_substatus_index)
+                self.__installation_packages_copy = self.__installation_substatus_msg_copy['patches']
+
+            status_file_without_package_list_size = self.__get_size_of_constant_status_data(copy.deepcopy(truncated_status_file), assessment_substatus_index, installation_substatus_index)  # deepcopy fully copy the object avoid reference modification
+            max_package_list_capacity = self.__internal_file_capacity - status_file_without_package_list_size
+            current_package_list_size = self.__get_current_packages_size_in_bytes(self.__assessment_packages_copy, self.__installation_packages_copy)
+
+            while status_file_size_in_bytes > self.__internal_file_capacity and current_package_list_size > max_package_list_capacity:
+                # Start truncation process
+                packages_truncated_in_assessment, packages_removed_from_assessment, packages_truncated_in_installation, packages_removed_from_installation = \
```

Comment:
why are they initialized with '\'?

Reply:
Q: Why are you not using self.__assessment_packages_removed directly instead of packages_removed_from_assessment?

Reply:
> nit: use retained instead of truncated
> 
> rename the variables to assessment_packages_retained_after_truncation, assessment_packages_removed_after_truncation, and similarly for installation lists

done

Reply:
> why are they initialized with '\'?

\ means newline, there are too many characters in that line

Reply:
> Q: Why are you not using self.__assessment_packages_removed directly instead of packages_removed_from_assessment?
it's a preference, if we have any values for packages_removed_from_assessment then i won't create a truncated status file and not initialize __assessment_packages_removed, then it will pick up by gc. using __assessment_packages_removed instead, then i need to initialize it to packages_removed_from_assessment, so it won't get pick up by gc